ggshield/pypi.py

```diff
@@ -0,0 +1,109 @@
+import re
+import shutil
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Dict, List, Set
+
+import click
+
+from ggshield.config import Config
+from ggshield.output import OutputHandler
+from ggshield.path import get_files_from_paths
+from ggshield.scan import Files, Result, ScanCollection
+from ggshield.utils import SupportedScanMode
+
+
+PYPI_DOWNLOAD_TIMEOUT = 30
+
+
+def save_package_to_tmp(temp_dir: str, package_name: str) -> None:
+    command: List[str] = [
+        "pip",
+        "download",
+        package_name,
+        "--dest",
+        temp_dir,
+        "--no-deps",
+    ]
+
+    try:
+        click.echo("Downloading pip package... ", nl=False)
+        subprocess.run(
+            command,
+            check=True,
+            stderr=subprocess.PIPE,
+            timeout=PYPI_DOWNLOAD_TIMEOUT,
+        )
+        click.echo("OK")
+
+    except subprocess.CalledProcessError:
+        raise click.ClickException(f'Package "{package_name}" not found')
```

Comment:
Download could fail for other reasons, like network issues. Maybe use a more generic message like `'Failed to download "{package_name}"'`?